include/OsmAndCore/Utilities.h

```diff
@@ -889,6 +889,38 @@ namespace OsmAnd
         static LatLon rhumbDestinationPoint(LatLon latLon, double distance, double bearing);
         static std::pair<int, int> calculateFinalXYFromBaseAndPrecisionXY(int bazeZoom, int finalZoom, int precisionXY,
                                                                           int xBase, int yBase, bool ignoreNotEnoughPrecision);
+        template<typename T>
+        inline static T computeDirection(
+            const T& start,
+            const T& end
+        )
+        {
+            auto length = std::sqrt((end.x - start.x)*(end.x - start.x) + (end.y - start.y)*(end.y - start.y));
+            return (end - start)/length;
+        }
+
+        template<typename T>
+        inline static T computeNormalToLine(
+            const T& start,
+            const T& end,
+            int dir
```

Comment:
bool as dir should be used here because dir is not used for computation only logical comparision